lib/screens_web/controllers/v2/screen_api_controller.ex

```diff
@@ -74,6 +74,23 @@ defmodule ScreensWeb.V2.ScreenApiController do
     end
   end
 
+  def simulation(conn, %{"id" => screen_id, "last_refresh" => last_refresh}) do
+    cond do
+      nonexistent_screen?(screen_id) ->
+        not_found_response(conn)
+
+      outdated?(screen_id, last_refresh) ->
+        json(conn, ScreenData.outdated_response())
```

Comment:
What's the behavior if we send this back to one of the simulation iframes running inside the screenplay app? Does only the iframe refresh, or does the whole containing screenplay page refresh?

Looks like since the reload logic runs _inside_ the iframe (i.e., in the screens client code, not screenplay), it should work the way we want it to—the `window` object will refer to the iframe in that context. However if that turns out to not be the case, we can address that by changing what we call in screens: https://stackoverflow.com/questions/86428/what-s-the-best-way-to-reload-refresh-an-iframe
([currently](https://github.com/mbta/screens/blob/master/assets/src/hooks/use_api_response.tsx#L86) we call `window.location.reload`)

Reply:
1. I think so—we should assume PIOs may keep a long-running tab open to screenplay, and make sure it doesn't get "out of sync" over time.
2. Hmm, that function should be able to return true. Every client loads with a timestamp of when it was initially loaded. It sends that timestamp with all of its data requests, and the server compares it with a timestamp kept in config for that screen ID. If we update the config with a later timestamp, then `outdated?/2` should return true and cause the server to send back a "please reload yourself" response

Reply:
Ahh, that admin tool request works a bit differently than I would have guessed. It adds a value into the config and then removes it once the refresh has occurred. So yeah, that's my bad.